### DISCUSSION TITLE
[SDK] Rust SDK

### DISCUSSION THREAD

**Discussion by 0xPetra** [zkSync-Community-Hub/zksync-developers#58] [SDKs] [answered]
### Environment

Testnet

### Acknowledgement

- [x] I understand that issues are expected with the Python and Java SDK as they are being updated.

### Issue Description

gm, I'm hacking a wallet with a Rust backend. 

Found this SDK that appears to not be fully finished by LambdaClass (https://github.com/lambdaclass/zksync-web3-rs). It uses a different `zksync-cli` to test locally, which I'm having trouble with. 

Also, does AA already work on this repo?

### Expected Behavior

Have a local instance of zkSync Era and integrate AA into a Rust backend.

### Code Example

This is the example provided to create the chain:

```
let private_key: Wallet<SigningKey> = "0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110".parse().unwrap();
let zksync_era_chain_id: u64 = 270;

let wallet = zksync::Wallet::with_chain_id(private_key, zksync_era_chain_id);
```

### Repo Link (Optional)

https://github.com/0xPetra/zkWallet

**Comment by uF4No** [comment, marked as answer]
hi @0xPetra That SDK is not officially released yet. As you mentioned LambdaClass is working on it. Feel free to create issues in the SDK repo directly.

The CLI you mention is a fork they created with an additional command to start/stop the dockerized local setup, but you can run that manually [as described in this section of the docs](https://era.zksync.io/docs/tools/hardhat/testing.html). Keep in mind the first time it runs, it can take ~10mins to deploy all system contracts locally.

**Comment by 0xPetra** [comment]
Great. Since they implemented the "up" command, wasn't sure there might be
something deployed specifically for the Rust SDK.

Already opened an issue.

Thanks Antonio!

On Tue, Jul 11, 2023 at 1:45 PM Antonio ***@***.***> wrote:

> hi @0xPetra <https://github.com/0xPetra> That SDK is not officially
> released yet. As you mentioned LambdaClass is working on it. Feel free to
> create issues in the SDK repo directly.
>
> The CLI you mention is a fork they created with an additional command to
> start/stop the dockerized local setup, but you can run that manually as
> described in this section of the docs
> <https://era.zksync.io/docs/tools/hardhat/testing.html>. Keep in mind the
> first time it runs, it can take ~10mins to deploy all system contracts
> locally.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zkSync-Community-Hub/zkync-developers/discussions/58#discussioncomment-6415483>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEZAN22TH4LMYGZWD54VQ4DXPU4GJANCNFSM6AAAAAA2F3BJH4>
> .
> You are receiving this because you were mentioned.Message ID:
> <zkSync-Community-Hub/zkync-developers/repo-discussions/58/comments/6415483
> @github.com>
>


-- 

<https://www.linhub.co/>

<http://twitter.com/0xPetra>
*Franco Petra*

*0xpetra.com <http://0xpetra.com> | **.eth
<https://etherscan.io/address/0x9fc3b33884e1d056a8ca979833d686abd267f9f8>*
* | **.
<https://etherscan.io/address/0x9fc3b33884e1d056a8ca979833d686abd267f9f8>lens*

**Comment by 0xPetra** [comment]
I just got this update from the LambdaClass team:

https://github.com/lambdaclass/zksync-web3-rs/issues/83#issuecomment-1631020190

I'm thinking about moving forward with the UI of the desktop wallet (in Rust) and integrating AA for Ethereum L1. 
When the SDK is ready, plan to integrate zkSync.

I wondered if this would still be a valid project submission for the BuidlEra hackathon.

**Comment by martinerinv** [comment]
It seems like you're facing challenges integrating zkSync into your Rust backend using the zksync-web3-rs SDK. Unfortunately, the SDK you mentioned might not be fully finished, which could lead to issues with testing locally using the provided zksync-cli.

As for the integration of zkSync into a Rust backend, there might be some difficulties due to the incomplete state of the SDK. Additionally, it's unclear if zkSync's zkSync accounts (AA) are fully supported in the current state of the repository.

To address these issues, you might consider reaching out to the maintainers of the zksync-web3-rs repository for support or exploring alternative SDKs or libraries that offer better support for zkSync integration into Rust backends. Additionally, ensuring that the zkSync chain ID is correctly set in your code can help with proper integration and testing.

**Comment by tysaboswe** [comment]
Issue: Difficulty testing locally with the zksync-web3-rs SDK and unsure if Atomic Arbitrum (AA) is supported.
Expected: Ability to easily test locally and integrate Atomic Arbitrum into a Rust backend.
SDK: https://github.com/lambdaclass/zksync-web3-rs

**Comment by sinanceviccarter** [comment]
To set up a local instance of zkSync Era and integrate with a Rust backend, you can utilize the zksync-web3-rs SDK by LambdaClass. However, ensure its compatibility with the latest version of zkSync. Regarding zkSync's Account Abstraction (AA) functionality, you may need to check the repository's documentation or contact the maintainers for information on its implementation status.

**Comment by lashawnheisler** [comment]
The provided SDK from LambdaClass is still under development and may not be fully functional. You may encounter issues with the zksync-cli tool when testing locally. Additionally, support for account abstraction (AA) may not be implemented yet in this repository. Make sure to check the latest updates and contributions to see if these features have been added.

**Comment by hyal1ngrumblu** [comment]
The user is attempting to integrate zkSync Era into a Rust backend for their wallet on the Testnet. They have found an SDK by LambdaClass (https://github.com/lambdaclass/zksync-web3-rs) but are experiencing issues with the different zksync-cli used for local testing. The user is also inquiring if AA already works on this repo. They have provided an example of the code they are using to create the chain. The user has not provided any error messages or logs. The user's repository link is https://github.com/0xPetra/zkWallet.
